server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2039,7 +2040,13 @@ private void innerOpenEngineAndTranslog(LongSupplier globalCheckpointSupplier) t
     }
 
     private boolean assertSequenceNumbersInCommit() throws IOException {
-        final Map<String, String> userData = SegmentInfos.readLatestCommit(store.directory()).getUserData();
+        final Map<String, String> userData;
+        if (isRemoteSnapshotIndex() && FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT_EXTENDED_BWC)) {
+            // Inefficient method to support reading old Lucene indexes
+            userData = Lucene.readAnySegmentInfos(store.directory()).getUserData();
+        } else {
+            userData = Lucene.readSegmentInfos(store.directory()).getUserData();
```

Comment:
why is the fallback here different & not `SegmentInfos.readLatestCommit(store.directory()).getUserData();` ?